File: leetcode/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int tmp = 1;
        for(int i = digits.size() - 1; i >= 0; --i){
            int curr = digits[i] + tmp;
            digits[i] = curr % 10;
            tmp = curr / 10;
        }
        if(tmp == 1){
            digits.insert(digits.begin(), 1);
        }
        return digits;
    }
};

File: leetcode/66. Plus One.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        step = 1
        for i in range(len(digits)-1, -1, -1):
            curr = digits[i] + step
            digits[i] = curr % 10
            step = curr // 10
        if step == 1:
            return [1] + digits
        else:
            return digits
